### PR TITLE
Reduce size of public API

### PR DESCRIPTION
**Pull request by fredrikdanielmoller**
Some structs and #defined constants were defined
in the public header file but not used for any
public API function.

By moving them to private header and source files,
the public API is made smaller and thus easier to
use.

Compile-time checks for the FILENAME
constants were also added.

See #243.

### CODE REVIEW

File: src/device/pf_fspm.c [hunk 1]
```diff
@@ -33,6 +33,14 @@
 #include "pf_includes.h"
 #include "pf_block_reader.h"
 
+typedef struct pnet_im_nvm
```
Comment (pyhys) [inline src/device/pf_fspm.c:36]: pf_im_nvm

File: src/pf_types.h [hunk 2]
```diff
@@ -2323,6 +2323,13 @@ typedef struct pf_interface_stats
    uint32_t if_out_errors;
 } pnet_interface_stats_t;
 
+typedef struct pnet_ieee_macphy_config
+{
+   uint8_t aneg_support_status; /**< Autonegotiation status */
+   uint16_t aneg_cap;           /**< Autonegotiation capabilites */
+   uint16_t operational_mau_type;
+} pnet_ieee_macphy_t;
```
Comment (pyhys) [inline src/pf_types.h:2331]: pf_ieee_macphy_t;

**Review by pyhys — COMMENTED** [review]
Great!
However I think the structs we move from pnet_api.h should be renamed pf_xxx.
See a few examples below.

File: src/device/pf_fspm.c (hunk 1 above)
Comment (fredrikdanielmoller) [reply src/device/pf_fspm.c]: Done (in separate Commit).

File: src/pf_types.h (hunk 2 above)
Comment (fredrikdanielmoller) [reply src/pf_types.h]: Done (in separate Commit).

**Comment by fredrikdanielmoller** [comment]
Rebased. Changed prefix pnet_ -> pf_ in separate Commit.

File: src/pf_types.h [hunk 3]
```diff
@@ -2464,6 +2495,46 @@ typedef struct pf_lldp_signal_delay
...
+typedef struct pf_lldp_peer_to_peer_boundary
+{
+   pf_lldp_boundary_t boundary;
+   uint16_t properites;
+} pf_lldp_peer_to_peer_boundary_t;
+
+/**
+ * LLDP Peer information used by the Profinet stack.
+ */
+typedef struct pft_lldp_peer_info
```
Comment (pyhys) [inline src/pf_types.h:2517]: `pf_lldp_peer_info`

Comment (fredrikdanielmoller) [reply src/pf_types.h]: Done.

**Review by pyhys — APPROVED** [review]
Great work!
This helps the end users by reducing the clutter.

**Comment by fredrikdanielmoller** [comment]
Rebase

**Review by olbjo — APPROVED** [review]
Great!

**Pull request merged**